### PR TITLE
Feature/tao 8511/refactor basic auth type

### PR DESCRIPTION
**Pull request by tikhanovichA**
https://oat-sa.atlassian.net/browse/TAO-8511
Depends on https://github.com/oat-sa/tao-core/pull/2214

### CODE REVIEW

File: model/bootstrap/OAuth2AuthType.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```
Comment (tikhanovichA) [inline model/bootstrap/OAuth2AuthType.php]:
```suggestion
 * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
```

File: test/model/Oauth2TypeTest.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            'client_secret' => 'client_secret',
+            'token_url' => 'token_url',
+            'token_type' => 'token_type',
+            'grant_type' => 'grant_type'
+        ];
+    }
+
+    public function testOAuth2Type()
+    {
+        $Oauth2Service = $this->createMock(Oauth2Service::class);
```
Comment (bartlomiejmarszal) [inline test/model/Oauth2TypeTest.php]:
```suggestion
        $Oauth2ServiceMock = $this->createMock(Oauth2Service::class);
```

**Comment by tikhanovichA** [comment]
> I think we should also create unit test for following logic:
> 
> ```
> if (!empty($credentials[Provider::TOKEN_TYPE])) {
>     $data[Provider::TOKEN_TYPE] = $credentials[Provider::TOKEN_TYPE];
> }
> ```
> 
> ```
> if (!empty($credentials[Provider::GRANT_TYPE])) {
>     $data[Provider::GRANT_TYPE] = $credentials[Provider::GRANT_TYPE];
> }
> ```
> 
> ```
> if (!empty($clientOptions)) {
>     $data[Provider::HTTP_CLIENT_OPTIONS] = $clientOptions;
> }
> ```

But there are not required fields. Why we should test them?

**Comment by bartlomiejmarszal** [comment]
> > I think we should also create unit test for following logic:
> > ```
> > if (!empty($credentials[Provider::TOKEN_TYPE])) {
> >     $data[Provider::TOKEN_TYPE] = $credentials[Provider::TOKEN_TYPE];
> > }
> > ```
> > 
> > 
> > ```
> > if (!empty($credentials[Provider::GRANT_TYPE])) {
> >     $data[Provider::GRANT_TYPE] = $credentials[Provider::GRANT_TYPE];
> > }
> > ```
> > 
> > 
> > ```
> > if (!empty($clientOptions)) {
> >     $data[Provider::HTTP_CLIENT_OPTIONS] = $clientOptions;
> > }
> > ```
> 
> But there are not required fields. Why we should test them?

Because we have some custom bahaviuor there. We want to ensure that this stays this same always. This is actually the only thing I would test here.

**Pull request merged**